### ISSUE TITLE
Fix contract checking for !null -> !null contracts (#1440)

### ISSUE THREAD

**Issue by giruuuuj** [uber/NullAway#1442]
## Problem
Contract checking was reporting false positives for !null -> !null contracts due to overly simplistic control flow analysis.

### Examples Fixed
All three variants from issue #1440 should now work correctly:

```java
@Contract("!null -> !null")
public static @Nullable Integer of(@Nullable String text) {
    if (text != null) {
        return Integer.parseInt(text);
    }
    else {
        return null;  // This should NOT trigger an error
    }
}

@Contract("!null -> !null")
public static @Nullable Integer of(@Nullable String text) {
    if (text != null) {
        return Integer.parseInt(text);
    }
    return null;  // This should NOT trigger an error
}

@Contract("!null -> !null")
public static @Nullable Integer of(@Nullable String text) {
    return (text != null) ? Integer.parseInt(text) : null;  // This should NOT trigger an error
}

**Comment by giruuuuj** [comment]
https://github.com/giruuuuj/NullAway

**Comment by msridhar** [comment]
@giruuuuj I already put up a fix for #1440 in #1441.  Also, this is an issue, not a pull request.
